idseq_dag/steps/run_srst2.py

```diff
@@ -0,0 +1,44 @@
+import os
+
+from idseq_dag.engine.pipeline_step import PipelineStep
+from idseq_dag.util.s3 import fetch_from_s3
+import idseq_dag.util.command as command
+
+class PipelineStepRunSRST2(PipelineStep):
+    '''
+    srst2 is used to detect resistance genes.
+    See: https://github.com/katholt/srst2
+    '''
+    def run(self):
+        ''' Invoking srst2 '''
+        db_file_path = fetch_from_s3(self.additional_files["resist_gene_db"], self.output_dir_local)
+        min_cov = str(self.additional_attributes['min_cov'])
+        n_threads = str(self.additional_attributes['n_threads'])
+        if len(self.input_files_local[0]) == 2:
+            fwd_rd = self.input_files_local[0][0]
+            rev_rd = self.input_files_local[0][1]
+            # Rename to format srst2 expects
+            command.execute(f"mv {fwd_rd} _R1_001.fastq.gz")
```

Comment:
instead of using mv, it's probably better to use ln -sf in case some other steps in the same pipeline also depend on the same input file.

Reply:
Also, don't you have to deal with the fastq files that are not compressed?

Reply:
first comment resolved, marked second as TODO to deal with fasta files